app/api/public-cloud/aws-roles/routes.ts

```diff
@@ -0,0 +1,155 @@
+import axios from 'axios';
+
+interface Group {
+  id: string;
+  name: string;
+  path: string;
+  subGroups: Group[];
+}
+
+interface User {
+  id: string;
+  username: string;
+  enabled: boolean;
+  totp: boolean;
+  emailVerified: boolean;
+  firstName: string;
+  lastName: string;
+  email: string;
+}
+
+const parseError = (error: unknown): void => {
+  if (error instanceof Error) {
+    console.log(error.message);
+  } else console.log(String(error));
+};
+const getToken = async (): Promise<string | undefined> => {
+  try {
+    const apiUrl = `${process.env.AWS_ROLES_BASE_URL}/realms/${process.env.AWS_ROLES_REALM_NAME}/protocol/openid-connect/token`;
+    const requestBody = {
+      client_id: process.env.AWS_ROLES_CLIENT_ID,
+      client_secret: process.env.AWS_ROLES_CLIENT_SECRET,
+      grant_type: 'client_credentials',
+    };
+    const response = await axios.post(apiUrl, requestBody, {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+    return response.data.access_token;
+  } catch (error: unknown) {
+    parseError(error);
+  }
+};
+
+const awsRolesApiInstance = axios.create({
+  baseURL: `${process.env.AWS_ROLES_BASE_URL}/admin/realms/${process.env.AWS_ROLES_REALM_NAME}`,
```

Comment:
I see the baseUrl is different than one used for `getToken`:
- `${process.env.AWS_ROLES_BASE_URL}/admin/realms/${process.env.AWS_ROLES_REALM_NAME}`
vs
- `${process.env.AWS_ROLES_BASE_URL}/realms/${process.env.AWS_ROLES_REALM_NAME}`
can you double check which one is the correct base url and also we might define it on top as a variable to avoid mis-sync.

Reply:
double checked, it is correct, they are different for get token and others